metricflow/query/query_parser.py

```diff
@@ -529,6 +537,34 @@ def _validate_order_by_specs(
             ):
                 raise InvalidQueryException(f"Order by item {order_by_spec} not in the query")
 
+    def _validate_date_part(
+        self, metric_references: Sequence[MetricReference], time_dimension_specs: Sequence[TimeDimensionSpec]
+    ) -> None:
+        """Validate that date parts can be used for metrics.
+
+        TODO: figure out expected behavior for date part with these types of metrics.
+        """
+        date_part_requested = False
+        for time_dimension_spec in time_dimension_specs:
+            if time_dimension_spec.date_part:
+                date_part_requested = True
+                if time_dimension_spec.date_part.to_int() < time_dimension_spec.time_granularity.to_int():
+                    raise RequestTimeGranularityException(
+                        f"Date part {time_dimension_spec.date_part.name} is not compatible with time granularity "
+                        f"{time_dimension_spec.time_granularity.name}."
```

Comment:
Shall we update this to indicate what is compatible? It's any date part that's bigger, right? So you can ask for year on daily granularity, but you can't ask for day on yearly granularity.

Reply:
Yes! I can add that!